opensearch-operator/pkg/reconcilers/tls.go

```diff
@@ -56,6 +56,12 @@ const (
 )
 
 func (r *TLSReconciler) Reconcile() (ctrl.Result, error) {
+	if r.instance.Spec.General.DisableSSL {
+		r.logger.Info("SSL is disabled. Skipping TLS setup")
+		r.reconcilerContext.AddConfig("plugins.security.disabled", "true")
```

Comment:
This disables the security plugin and it is much more than disabling SSL.

If the original issue only requires to disable REST layer TLS(which users interact mainly), then we need to set `plugins.security.ssl.http.enabled` as `false`. Its default value is `false` and operator sets it as `true` here. https://github.com/opensearch-project/opensearch-k8s-operator/blob/3372c11314032bc618f109cc8eded0f2215178a3/opensearch-operator/pkg/reconcilers/tls.go#L540

There is no option to disable TLS for transport layer yet. https://github.com/opensearch-project/security/issues/4759